src/groups/semidirect_product_group.jl

```diff
@@ -392,89 +969,97 @@ function get_vector_lie!(
+    ) where {𝔽, O1, O2, A <: AbstractLeftGroupActionType, AO <: AbstractActionActsOnType}
+    PM, G, H, a, g_ind, h_ind = _semidirect_parts(SDPG)
+    # (a) compute the inverse in G
+    inv!(G, submanifold_component(SDPG, k, Val(g_ind)), submanifold_component(PM, g, Val(g_ind)))
+    # (b) compute the inverse in H
+    inv!(H, submanifold_component(SDPG, k, Val(h_ind)), submanifold_component(PM, g, Val(h_ind)))
+    # (c) apply the group action w.r.t. the inverse in G (from a) to the inverse from (b)
+    apply!( # Apply the group action with g1^-1 to g2^-1
+        a,
+        submanifold_component(SDPG, k, Val(h_ind)),
+        submanifold_component(SDPG, k, Val(g_ind)),
+        submanifold_component(SDPG, k, Val(h_ind)),
+        #TODO make sure apply is safe against side effects
     )
     return k
 end
-function inv!(
-        SDPG::LieGroup{𝔽, O, M}, k, ::Identity{O}
-    ) where {𝔽, O <: RightSemidirectProductGroupOperation, M <: ProductManifold}
-    PrM = SDPG.manifold
-    map(
-        inv!,
-        map(LieGroup, PrM.manifolds, SDPG.op.operations),
-        submanifold_components(PrM, k),
-        map(Identity, SDPG.op.operations),
+
+# 3. Left semidirect, right action, act on left
+# 4. Left semidirect, right action, act on right
+# 8. Right semidirect, right action, act on right
+# 7. Right semidirect, right action, act on left
+"""
+    inv(L::LieGroup{𝔽,<:SemidirectProductGroupOperation{⋆,⋄,<:AbstractRightGroupActionType,AbstractActionActsOn}}, g)
+
+$(_doc_semidirect_sub_groups) Let ``τ`` denote a right group action. The inverse is the same for both [`ActionActsOnLeft`](@ref) and [`ActionActsOnRight`](@ref).
+
+The inverse for the [`LeftSemidirectProductGroupOperation`](@ref) ``$(_math(:∘))`` on ``G ⋉ H`` is given by
+
+```math
+(g,h)^{-1} = $(_tex(:bigl))( g^{-1}, τ_{g}(h^{-1}) $(_tex(:bigr))).
+```
+
+The inverse for the [`RightSemidirectProductGroupOperation`](@ref) ``$(_math(:∘))`` on ``G ⋊ H`` is given by
+
+```math
+(h,g)^{-1} = $(_tex(:bigl))( τ_{g}(h^{-1}), g^{-1} $(_tex(:bigr)))
+```
+"""
+inv(
+    SDPG::LieGroup{𝔽, <:SemidirectProductGroupOperation{O1, O2, A, AO}, <:ProductManifold}, ::Any,
+) where {𝔽, O1, O2, A <: AbstractRightGroupActionType, AO <: AbstractActionActsOnType}
+
+function _inv!(
+        SDPG::LieGroup{𝔽, <:SemidirectProductGroupOperation{O1, O2, A, AO}, <:ProductManifold}, k, g
+    ) where {𝔽, O1, O2, A <: AbstractRightGroupActionType, AO <: AbstractActionActsOnType}
+    PM, G, H, a, g_ind, h_ind = _semidirect_parts(SDPG)
+    # (a) compute the inverse in H
+    inv!(H, submanifold_component(SDPG, k, Val(h_ind)), submanifold_component(PM, g, Val(h_ind)))
+    # (b) apply the group action w.r.t. g in G (not yet inverted in place) to the inverse from (b)
+    apply!( # Apply the group action with g1^-1 to g2^-1
+        a,
+        submanifold_component(SDPG, k, Val(h_ind)),
+        submanifold_component(SDPG, k, Val(g_ind)),
```

Comment:
I think this is a typo and should maybe be from g
```suggestion
        submanifold_component(SDPG, g, Val(g_ind)),
```

Reply:
Thanks for spotting this is correct. Assume we have the left semidirect one, then (a) computed g_2^-1 (and stores it in k_2) – (b) applies g_1*k_2 (or g_1g_2^-1) and stores it in k_2 as well. (c) computes g_1^-1 and stores it in k_1

(c) has to be done ofter (b) to avoid aliased effects.